### PR TITLE
Split Razor extensibility into its own assembly

### PR DESCRIPTION
**Pull request by rynowak**
This functionality will need to cross-compile to desktop framework
(net461) so that we can use it in VS. VS doesn't yet have netstandard2.0
support.

### CODE REVIEW

**Comment by rynowak** [comment]
No really cool changes here, just separating the Razor stuff into its own project so it can be included in VS. This helps prevent code rot in my VS prototyping work.

**Comment by rynowak** [comment]
The next step for this will be to introduce latest builds of Razor and start checking in actual working VS bits. Let's discuss this at tomorrow's sync up.

File: src/Microsoft.AspNetCore.Blazor.Razor.Extensions/RenderTreeBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+        public static readonly string OpenComponent = nameof(OpenComponent);
+
+        public static readonly string CloseComponent = nameof(CloseElement);
+
+        public static readonly string AddText = nameof(AddText);
+
+        public static readonly string AddAttribute = nameof(AddAttribute);
+        
+        public static readonly string Clear = nameof(Clear);
```
Comment (SteveSandersonMS) [inline src/Microsoft.AspNetCore.Blazor.Razor.Extensions/RenderTreeBuilder.cs]: I suspect we won't need `Clear` or `GetFrames` here, as they would never be used in the codegenned code.

File: test/Microsoft.AspNetCore.Blazor.E2ETest/Infrastructure/BrowserFixture.cs [hunk 2]
```diff
@@ -16,7 +16,18 @@ public BrowserFixture()
...
+            try
+            {
+                Browser = new RemoteWebDriver(opts);
+            }
+            catch (WebDriverException ex)
+            {
+                var message =
+                    "Failed to connect to the web driver. Please see the readme and follow the instructions to install selenium." +
+                    "Remember to start the web driver with `selenium-standalone start` before running the end-to-end tests.";
+                throw new InvalidOperationException(message, ex);
```
Comment (SteveSandersonMS) [inline test/Microsoft.AspNetCore.Blazor.E2ETest/Infrastructure/BrowserFixture.cs]: This is better than the uninformative error we had before.  Later on we could look into launching and stopping selenium server automatically from the tests. This is fine for now though :)

**Pull request merged**